pkg/solana/txm/pendingtx.go

```diff
@@ -271,6 +277,30 @@ func (c *pendingTxContext) OnProcessed(sig solana.Signature) (string, error) {
 	})
 }
 
+// IsBlockhashExpired returns whether or not the blockhash of a transaction has expired when compared to currHeight.
+func (c *pendingTxContext) IsBlockhashExpired(id string, currHeight uint64) bool {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+	tx, exists := c.broadcastedTxs[id]
+	if !exists {
+		return false
+	}
+	return tx.lastValidBlockHeight < currHeight
```

Comment:
I think we would want to also double check the `tx.state` to make sure it's `Broadcasted`. I probably could have done a better job naming this map but transactions in here could also be `Processed` in which case maybe expiration doesn't matter